src/MidiPlayer.js

```diff
@@ -624,4 +625,36 @@ not.
         this.logging = logging;
         this.eventHandler.setLogger({ eventLogger, logging });
     }
+    /*
+     * get total duration time
+     * 2021 03 NormalUniversity
+     */
+    getDuration() {
+
+        return this.totalTime;
+    }
+    /**
+     * Seek Position.
+     * @param {any} [mSecond] milli seconds to be seek
+     * @example
+     *  let midiPlayer = new MidiPlayer({ logging: false, patchUrl: 'lib/pat/' });
+     *  midiPlayer.play({url: $url})
+     *  midiPlayer.seek({mSecond: $milliSeconds});
+     */
+    async seek({ mSecond}) {
+        if (this.totalTime>0) {
+            let _mSecond = this.sampleRate * mSecond;
+            LibTiMidity.call('mid_song_seek', 'void', ['number', 'number'], [this.song, _mSecond]);
+            try {
```

Comment:
It might be a good idea to include the LibTimidity call into the `try` block in order to catch errors that could occur when seeking.